website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -362,9 +363,44 @@ api.approveTask = {
       direction,
     });
 
-    managerPromises.push(task.save());
-    managerPromises.push(assignedUser.save());
-    await Promise.all(managerPromises);
+    // Handle shared completion
+    let masterTask = await Tasks.Task.findOne({
+      _id: taskId,
+    }).exec();
+
+    if (task.type === 'todo' && masterTask && masterTask.group && masterTask.group.sharedCompletion === 'singleCompletion') {
```

Comment:
if this code is the same as `website/server/controllers/api-v3/tasks.js` maybe it should be moved to a library file